compiler/src/linking/link.re

```diff
@@ -0,0 +1,529 @@
+          let internal_name = Function.get_name(func);
+          let new_name =
+            Hashtbl.find(
+              Hashtbl.find(exported_names, imported_module),
+              imported_name,
+            );
+          Hashtbl.add(local_names, internal_name, new_name);
+        } else {
+          let imported_name = Import.function_import_get_base(func);
+          let internal_name = Function.get_name(func);
+          let new_name = gensym(internal_name);
+          Hashtbl.add(local_names, internal_name, new_name);
+
+          let params = Function.get_params(func);
+          let results = Function.get_results(func);
+          Import.add_function_import(
+            linked_mod,
+            new_name,
+            imported_module,
+            imported_name,
+            params,
+            results,
+          );
+        };
+      },
+      imported_funcs,
+    );
+
+    List.iter(
+      func => {
+        let internal_name = Function.get_name(func);
+        let new_name = gensym(internal_name);
+        Hashtbl.add(local_names, internal_name, new_name);
+
+        let params = Function.get_params(func);
+        let results = Function.get_results(func);
+        let num_locals = Function.get_num_vars(func);
+        let locals = Array.init(num_locals, i => Function.get_var(func, i));
+        let body = Function.get_body(func);
+        globalize_names(local_names, body);
+        ignore @@
+        Function.add_function(
+          linked_mod,
+          new_name,
+          params,
+          results,
+          locals,
+          body,
+        );
+      },
+      funcs,
+    );
+
+    let num_exports = Export.get_num_exports(wasm_mod);
+    let local_exported_names: Hashtbl.t(string, string) = Hashtbl.create(10);
+    for (i in 0 to num_exports - 1) {
+      let export = Export.get_export_by_index(wasm_mod, i);
+      let export_kind = Export.export_get_kind(export);
+      if (export_kind == Export.external_function
+          || export_kind == Export.external_global) {
+        let exported_name = Export.get_name(export);
+        let internal_name = Export.get_value(export);
+        let new_internal_name = Hashtbl.find(local_names, internal_name);
+        Hashtbl.add(local_exported_names, exported_name, new_internal_name);
+      };
+    };
+    Hashtbl.add(exported_names, dep, local_exported_names);
+
+    let num_element_segments = Table.get_num_element_segments(wasm_mod);
+    for (i in 0 to num_element_segments - 1) {
+      let segment = Table.get_element_segment_by_index(wasm_mod, i);
+      let name = Table.element_segment_get_name(segment);
+      let new_name = gensym(name);
+      let size = Table.element_segment_get_length(segment);
+      let elems =
+        List.init(size, i =>
+          Hashtbl.find(
+            local_names,
+            Table.element_segment_get_data(segment, i),
+          )
+        );
+      ignore @@
+      Table.add_active_element_segment(
+        linked_mod,
+        "tbl",
```

Comment:
I've seen this string a couple times in this file. Should it be a constant at the top?